### PR TITLE
Fix  Wrong Categories in Shortcuts Mapper

### LINKED ISSUE #13285: Wrong Categories in Shortcuts Mapper
**STR:**
Settings -> Shortcuts Mapper.

**Result:**
Some commands are listed under the wrong Category.
A few examples:

תמונה
תמונה
תמונה

***

Notepad++ v8.4.9   (64-bit)
Build time : Feb 24 2023 - 10:04:40
Path : C:---notepad++.exe
Command Line : 
Admin mode : ON
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit)
OS Version : 20H2
OS Build : 19042.985
Current ANSI codepage : 1255
Plugins : 
    ComparePlus (1.1)
    HTMLTag (1.3.6)
    MenuIcons (2.0.2)
    PythonScript (2)
    _CustomizeToolbar (4.2)

### PR DESCRIPTION
**Pull request by donho**
Fix #13285

### CODE REVIEW

**Comment by donho** [comment]
@Yaron10 
Could you check if it's fixed completely?

**Comment by Yaron10** [comment]
@donho,

Great. 👍 
Thank you for your work.

***
תמונה

Wouldn't it be better to associate those two commands with View?
***

תמונה

Still Setting instead of Settings.
***

Shouldn't the XML localization files be updated?

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 1]
```diff
@@ -1227,31 +1227,43 @@ intptr_t CALLBACK ScintillaKeyMap::run_dlgProc(UINT Message, WPARAM wParam, LPAR
...
+		_category = TEXT("View");
+	else if (_id == IDM_EDIT_LTR || _id == IDM_EDIT_RTL)
+		_category = TEXT("View");
+	else if (_id == IDM_FORMAT_TODOS || _id == IDM_FORMAT_TOUNIX || _id == IDM_FORMAT_TOMAC)
+		_category = TEXT("Edit");
+	else if (_id == IDM_EDIT_AUTOCOMPLETE || _id == IDM_EDIT_AUTOCOMPLETE_CURRENTFILE || _id == IDM_EDIT_FUNCCALLTIP ||
+		_id == IDM_EDIT_AUTOCOMPLETE_PATH || _id == IDM_EDIT_FUNCCALLTIP_PREVIOUS || _id == IDM_EDIT_FUNCCALLTIP_NEXT)
+		_category = TEXT("Edit");
+	else if (_id == IDM_LANGSTYLE_CONFIG_DLG)
+		_category = TEXT("Setting");
```
Comment (Yaron10) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp:1243]: Settings.

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 2]
```diff
@@ -1227,31 +1227,43 @@ intptr_t CALLBACK ScintillaKeyMap::run_dlgProc(UINT Message, WPARAM wParam, LPAR
...
 	else if ( _id < IDM_ABOUT)
-		_category = TEXT("Lang");
+		_category = TEXT("Language");
 	else if ( _id < IDM_SETTING)
 		_category = TEXT("About");
 	else if ( _id < IDM_TOOL)
-		_category = TEXT("Setting");
+		_category = TEXT("Settings");
 	else if ( _id < IDM_EXECUTE)
 		_category = TEXT("Tool");
```
Comment (Yaron10) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp:1266]: Tools.

**Comment by Yaron10** [comment]
@donho,

All commands are under the correct category now. 👍 
Thank you again.

Could you please explain to me the purpose of the last two commits?
The categories _were localized_ before them.

The only change needed was replacing:
```xml
				<WindowCategory name="Window"/>
				<FileCategory name="File"/>
				<EditCategory name="Edit"/>
				<SearchCategory name="Search"/>
				<ViewCategory name="View"/>
				<FormatCategory name="Format"/>
				<LangCategory name="Lang"/>
				<AboutCategory name="About"/>
				<SettingCategory name="Setting"/>
				<ToolCategory name="Tool"/>
				<ExecuteCategory name="Execute"/>
```
with
```xml
				<WindowCategory name="Window"/>
				<FileCategory name="File"/>
				<EditCategory name="Edit"/>
				<SearchCategory name="Search"/>
				<ViewCategory name="View"/>
				<EncodingCategory name="Encoding"/>
				<LanguageCategory name="Language"/>
				<AboutCategory name="About"/>
				<SettingsCategory name="Settings"/>
				<ToolsCategory name="Tools"/>
				<RunCategory name="Run"/>
```

Getting the categories from the main menu might be a good idea.
But then - the code getting them from `XCategory name` could and should be removed, shouldn't it?

**Comment by Yaron10** [comment]
@donho,

Thank you for the explanation.

> All the "*Category" strings are not using. I've checked in the codes - so they were not localized for sure.

What do you mean?

Here is v8.5.7.
תמונה

And this is after removing `<FileCategory name="קובץ"/>` from the localization file.
תמונה

**Comment by Yaron10** [comment]
[Make categories in the Shortcut Mapper dialog translatable](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/f7fcab4c219b7187ce9d6ca300c96f06f8f2e877).

**Comment by donho** [comment]
> [Make categories in the Shortcut Mapper dialog translatable](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/f7fcab4c219b7187ce9d6ca300c96f06f8f2e877).

Thank you for reminding. However, it doesn't work for me in v8.5.7

> And this is after removing <FileCategory name="קובץ"/> from the localization file.

Even with the whole PR ? I have tested it though. It does work on my PC.
Surely we don't deliver the PR with my PC...

**Comment by Yaron10** [comment]
@donho,

> Even with the whole PR ?

I was referring to v8.5.7 (as a proof that those strings do work).
With this PR the categories are localized after removing the "*Category" strings (Except About).

> However, it doesn't work for me in v8.5.7.

Very interesting. It does work for me on Windows 10 & Windows 7.
Any idea what it can be? Did you try it with a fresh portable NPP? :)

@alankilborn, @rdipardo, @rddim,
Are the categories localized on your machine with v8.5.7?

**Comment by xomx** [comment]
> Are the categories localized on your machine with v8.5.7?

Yes, they are (Win10, Win11).

**Comment by Yaron10** [comment]
@xomx,

Thank you!

**Comment by donho** [comment]
OK, I see why it didn't work for me.
On debug mode `doLocalConf.xml` is present, and there were already 3 obsolete localization files (English, French & Mandarin) in the localization folder beside the binary, whereas I have copied these 3 updated files into %APPDATA%\Notepad++\ - the debug binary load the obsolete localization (in which "*Category" entries are absent.

**Comment by donho** [comment]
But while I searched "WindowCategory", "FileCategory"... etc in the code, I found nothing - which was kind of obscure IMO.
And since we can get these translated string from the menu directly, why should we store them in localization XML file?
Hence the following commits.

**Comment by Yaron10** [comment]
@donho,

> And since we can get these translated string from the menu directly, why should we store them in localization XML file?

This is a question of an amateur to a master. :)
Getting the strings from the menu should apparently be a good choice for two reasons:
1. No double entries in the XML localization files.
2. The code _apparently_ should be shorter.

The code with this PR is longer and seems more complex.
Do you still think it's worth it?

Thank you.

**Comment by Yaron10** [comment]
@donho,

Good points. I trust your judgement. 👍 
Thank you again for your time and work.

**Comment by donho** [comment]
@Yaron10 
Thank you for your review!

**Pull request closed without merging**